examples/parallel_bert.py

```diff
@@ -0,0 +1,192 @@
+import os
+import argparse
+import functools
+import pandas as pd
+import pathlib
+import numpy as np
+import urllib.request
+import zipfile
+
+import torch
+import torch.nn as nn
+from torch.optim import Adam
+from torch.distributed.fsdp.wrap import transformer_auto_wrap_policy
+from torch.distributed.fsdp import BackwardPrefetch
+
+from gdeep.search.hpo import GiottoSummaryWriter
+from gdeep.data.datasets import FromArray, DataLoaderBuilder
+from gdeep.trainer.trainer import Trainer, Parallelism
+import gdeep.utility_examples.args
+
+from sklearn.model_selection import train_test_split
+from transformers import (
+                          BertTokenizer,
+                          BertForSequenceClassification,
+                          )
+from transformers.models.bert.modeling_bert import BertLayer
+
+
+def download_dataset():
+    if not pathlib.Path("cola_public").exists():
+        req = urllib.request.urlretrieve("https://nyu-mll.github.io/CoLA/cola_public_1.1.zip")
+        with zipfile.ZipFile(req[0], "r") as zip_ref:
+            zip_ref.extractall()
+
+
+def main(args):
+    n_sentences_to_consider=4000
+
+    tmp_path=os.path.join('./cola_public','raw','in_domain_train.tsv')
+    df = pd.read_csv(tmp_path, delimiter='\t', header=None, names=['sentence_source', 'label', 'label_notes', 'sentence'])
```

Comment:
put a come after names=[...]. This will made the line more readable

Reply:
@VascoSch92 Could you please explain what you mean exactly?

Reply:
Instead of writing
```python
df = pd.read_csv(tmp_path, delimiter='\t', header=None, names=['sentence_source', 'label', 'label_notes', 'sentence'])
```
use
```python
df = pd.read_csv(
    tmp_path, 
    delimiter='\t', 
    header=None, 
    names=['sentence_source', 'label', 'label_notes', 'sentence'],
)
```
Note the comma at the end of names. If you are using an automatic formatter (like yapf or black) and you put a comma at the end, it will format that for you.

The second one is much easier to read. You see exactly which method/class/function are you using and what are the parameters.

Reply:
Thanks for clarificaton @VascoSch92 . We have a pre-commit hook that runs black formatter.

@yorickbrunet Could you please use that? See here: https://github.com/giotto-ai/giotto-deep?tab=readme-ov-file#contributing